integration/client/go.mod

```diff
@@ -1,40 +0,0 @@
-module github.com/containerd/containerd/integration/client
```

Comment:
I think this `go.mod` can be kept, as this code is not for external consumption, and won't be release (it's just to keep integration test dependencies out of the main module's dependency tree)

Reply:
Updated to keep `integration/client` in since there is no other way to avoid vendoring in the shared object file (#5101)